### PR TITLE
Add feedback comments and suggestions

### PR DESCRIPTION
**Pull request by michaelrevans**
Here are a few small improvement suggestions that I have - just look at the diff in the code, it's all in comments/code changes

N.B if you merge this, the code won't work anymore, I made some changes without fully following through with them, so don't merge if you want this project to continue working

### CODE REVIEW

File: src/Provider.tsx [hunk 1]
```diff
@@ -15,7 +18,7 @@ interface State {
   loading: boolean,
   errors?: any,
   fetchRepos: ((term: string) => Promise<any>),
-  fetchRepo: ((owner: string, repo: string) => Promise<any>) 
+  fetchRepo: ((owner: string, repo: string) => Promise<any>)
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
  fetchRepo: ((owner: string, repo: string) => Promise<any>) 
```

File: src/Provider.tsx [hunk 2]
```diff
@@ -34,17 +37,17 @@ const initialState:State = {
   fetchRepo: (owner: string, repo: string) => new Promise((resolve, reject) => {}),
 }
 
-// reducer to handle global state change 
+// reducer to handle global state change
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
// reducer to handle global state change 
```

File: src/Provider.tsx [hunk 3]
```diff
@@ -34,17 +37,17 @@ const initialState:State = {
...
 const reducer: React.Reducer<State, Action> = (state, action) => {
   switch (action.type) {
     case ActionType.MAKING_API_REQUEST:
-      return { 
+      return {
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
      return { 
```

File: src/Provider.tsx [hunk 4]
```diff
@@ -34,17 +37,17 @@ const initialState:State = {
...
         ...state,
         loading: true,
         errors: {}
       }
     case ActionType.FAILED_API_REQUEST:
-      return { 
+      return {
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
      return { 
```

File: src/Provider.tsx [hunk 5]
```diff
@@ -53,19 +56,19 @@ const reducer: React.Reducer<State, Action> = (state, action) => {
         repoCount: null
       }
     case ActionType.SUCCESSFUL_REPOS_FETCH:
-      return { 
+      return {
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
      return { 
```

File: src/Provider.tsx [hunk 6]
```diff
@@ -53,19 +56,19 @@ const reducer: React.Reducer<State, Action> = (state, action) => {
...
         ...state,
         loading: false,
         errors: {},
         repos: action.payload.items,
         repoCount: action.payload.total_count
       }
     case ActionType.SUCCESSFUL_REPO_FETCH:
-      return { 
+      return {
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
      return { 
```

File: src/Provider.tsx [hunk 7]
```diff
@@ -53,19 +56,19 @@ const reducer: React.Reducer<State, Action> = (state, action) => {
...
         ...state,
         loading: false,
         errors: {},
-        repo: action.payload 
+        repo: action.payload
```
Comment (michaelrevans) [inline src/Provider.tsx]:
```suggestion
        repo: action.payload 
```

File: src/components/Search.tsx [hunk 8]
```diff
@@ -40,17 +48,20 @@ const Search: React.FC = () => {
...
             placeholder='Enter search term'
             onChange={ handleOnChange }
           />
         </div>
         <div>
           <p>Filter by language:</p>
-          <input 
-            type='text' 
+          <input
+            type='text'
```
Comment (michaelrevans) [inline src/components/Search.tsx]:
```suggestion
          <input 
            type='text' 
```

File: src/pages/Repos/Repo.tsx [hunk 9]
```diff
@@ -41,12 +46,14 @@ const Repo: React.FC = () => {
     })
     return repoKeys.map((k:any, i:number) => (
       <DataRow key={ k }>
-        <DataCell><strong>{ k }: </strong></DataCell> 
+        <DataCell><strong>{ k }: </strong></DataCell>
```
Comment (michaelrevans) [inline src/pages/Repos/Repo.tsx]:
```suggestion
        <DataCell><strong>{ k }: </strong></DataCell> 
```

File: src/pages/Repos/Repo.tsx [hunk 10]
```diff
@@ -41,12 +46,14 @@ const Repo: React.FC = () => {
...
         <DataCell>{ repoValues[i] }</DataCell>
       </DataRow>
-    ))      
+    ))
```
Comment (michaelrevans) [inline src/pages/Repos/Repo.tsx]:
```suggestion
    ))      
```

File: src/pages/Repos/Repo.tsx [hunk 11]
```diff
@@ -61,19 +68,21 @@ const Repo: React.FC = () => {
           <p>Open Issues: { repo?.open_issues }</p>
           <p>Watchers: { repo?.watchers }</p>
           <p>Default Branch: { repo?.default_branch }</p>
-          </DetailRow> 
+          </DetailRow>
```
Comment (michaelrevans) [inline src/pages/Repos/Repo.tsx]:
```suggestion
          </DetailRow> 
```

File: src/pages/Repos/Repo.tsx [hunk 12]
```diff
@@ -61,19 +68,21 @@ const Repo: React.FC = () => {
...
+  // memoizing these two - if the repo doesn't change, neither will these values
   const repoDetails = repo ? parseRepoDetails(repo) : null
   const truncatedRepo = repo ? truncatedDetails(repo) : null
-     
+
   return (
     <DetailPage>
       <NavLink to='/'>Back to search results</NavLink>
-      { showAllDetails ? 
+      { showAllDetails ?
```
Comment (michaelrevans) [inline src/pages/Repos/Repo.tsx]:
```suggestion
      { showAllDetails ? 
```

**Review by michaelrevans — COMMENTED** [review]
remove some trailing whitespace diffs

**Comment by abadfish** [comment]
Michael, 
Thank you so much for your perspective and suggestions! I really appreciate you taking the time to review my code and offer detailed suggestions. If I can ever return the favor, please don't hesitate to ask :)

**Pull request closed without merging**